### ISSUE TITLE
大佬 请问八连接聚类模块是哪里呀，我好像没找到对应的代码，还有好像模型中有一行代码的模块没有用到前向传播函数中

### ISSUE THREAD

**Issue by wqdebi** [TianhaoWu16/Multi-level-TransUNet-for-Space-based-Infrared-Tiny-ship-Detection#12]
# self.final2 = nn.BatchNorm2d(num_classes)
        self.final = nn.Conv2d(nb_filter[0], num_classes, kernel_size=1)
        # for name,value in self.final.named_parameters():
        #     value.requires_grad = False

**Comment by TianhaoWu16** [comment]
关于聚类的算法，在计算IOU指标和PD Fa指标的代码中有。
你好你提的这行是最终分类层，因为我们的工作里是二分类，是在传播过程中的，是函数的最后一层。



---原始邮件---
发件人: ***@***.***&gt;
发送时间: 2024年12月6日(周五) 下午5:18
收件人: ***@***.***&gt;;
抄送: ***@***.***&gt;;
主题: [TianhaoWu16/Multi-level-TransUNet-for-Space-based-Infrared-Tiny-ship-Detection] 大佬 请问八连接聚类模块是哪里呀，我好像没找到对应的代码，还有好像模型中有一行代码的模块没有用到前向传播函数中 (Issue #12)




     # self.final2 = nn.BatchNorm2d(num_classes)     self.final = nn.Conv2d(nb_filter[0], num_classes, kernel_size=1)     # for name,value in self.final.named_parameters():     #     value.requires_grad = False

**Comment by wqdebi** [comment]
太感谢了大佬，没想到能得到您怎么快的回复，希望您能在科研事业上取得更大的成果

**Comment by wqdebi** [comment]
> 关于聚类的算法，在计算IOU指标和PD Fa指标的代码中有。 你好你提的这行是最终分类层，因为我们的工作里是二分类，是在传播过程中的，是函数的最后一层。
> […](#)
> ---原始邮件--- 发件人: ***@***.***&gt; 发送时间: 2024年12月6日(周五) 下午5:18 收件人: ***@***.***&gt;; 抄送: ***@***.***&gt;; 主题: [TianhaoWu16/Multi-level-TransUNet-for-Space-based-Infrared-Tiny-ship-Detection] 大佬 请问八连接聚类模块是哪里呀，我好像没找到对应的代码，还有好像模型中有一行代码的模块没有用到前向传播函数中 (Issue #12) # self.final2 = nn.BatchNorm2d(num_classes) self.final = nn.Conv2d(nb_filter[0], num_classes, kernel_size=1) # for name,value in self.final.named_parameters(): # value.requires_grad = False — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you are subscribed to this thread.Message ID: ***@***.***&gt;

太感谢了大佬，没想到能得到您怎么快的回复，希望您能在科研事业上取得更大的成果

**Comment by wqdebi** [comment]
> > 关于聚类的算法，在计算IOU指标和PD Fa指标的代码中有。 你好你提的这行是最终分类层，因为我们的工作里是二分类，是在传播过程中的，是函数的最后一层。
> > […](#)
> > ---原始邮件--- 发件人: _**@**_._**> 发送时间: 2024年12月6日(周五) 下午5:18 收件人: _**@**_.**_>; 抄送: _**@**_._**>; 主题: [TianhaoWu16/Multi-level-TransUNet-for-Space-based-Infrared-Tiny-ship-Detection] 大佬 请问八连接聚类模块是哪里呀，我好像没找到对应的代码，还有好像模型中有一行代码的模块没有用到前向传播函数中 (Issue #12) # self.final2 = nn.BatchNorm2d(num_classes) self.final = nn.Conv2d(nb_filter[0], num_classes, kernel_size=1) # for name,value in self.final.named_parameters(): # value.requires_grad = False — Reply to this email directly, view it on GitHub, or unsubscribe. You are receiving this because you are subscribed to this thread.Message ID: _**@**_.**_>
> 
> 太感谢了大佬，没想到能得到您怎么快的回复，希望您能在科研事业上取得更大的成果

我好像打错了，是这一层没有用到
self.head = _FCNHead(nb_filter[0], channels=num_classes, momentum=0.9)
